wgpu-core/src/device/mod.rs

```diff
@@ -117,6 +117,9 @@ fn map_buffer<B: hal::Backend>(
     sub_range: hal::buffer::SubRange,
     kind: HostMap,
 ) -> BufferMapResult {
+    if let resource::BufferMapState::Active = buffer.map_state {
+        log::error!("Buffer already mapped");
```

Comment:
let's panic or return an error instead? proceeding seems like it would confuse the internal logic anyway and cause to panics later down the road